feat: add multi-org generate_params.py (#672)
awslabs/aws-deployment-framework: docs/user-guide.md

deployment map as a target. If only the account number is in the deployment map
the corresponding OU parameter file will not be referenced.

> 
> 

Reviewer: Could you remove these empty lines, so there is only one around this code block?

```suggestion
```

*Note:* Currently only Strings type values are supported as parameters to
CloudFormation templates when deploying via AWS CodePipeline.

#### CloudFormation Parameters in a Multi-Organization ADF Setup

> The CloudFormation Parameter generation feature is fully compatible with 
> the [Multi-Organization ADF Setup](./multi-organization-guide.md) approach. 
> 
> For example, in a setup with three AWS Organizations; with a separate
> `dev`, an `int`, and a `prod` AWS Organization. This implies that the
> SSM parameter `/adf/org/stage` will have one of the following three
> values: `dev`, `int`, or `prod`; depending on the AWS organization 
> you are in. Let's further assume that your application in scope 
> requires AWS Organization specific parameters. In that case,
> the `params` folder should  have the following content: 

Reviewer: Suggestion fixes trailing white spaces linting error:

```suggestion
The CloudFormation Parameter generation feature is fully compatible with
the [Multi-Organization ADF Setup](./multi-organization-guide.md) approach.

For example, in a setup with three AWS Organizations; with a separate
`dev`, an `int`, and a `prod` AWS Organization. This implies that the
SSM parameter `/adf/org/stage` will have one of the following three
values: `dev`, `int`, or `prod`; depending on the AWS organization
you are in. Let's further assume that your application in scope
requires AWS Organization specific parameters. In that case,
the `params` folder should  have the following content:
```

The dead link issue will be resolved if we merge PR #668 first.

*Note:* Currently only Strings type values are supported as parameters to
CloudFormation templates when deploying via AWS CodePipeline.

> **CloudFormation Parameters in a Multi-Organization ADF Setup** 

Reviewer: Could you change this to a sub section, like so?
That way we can reference it in GitHub issues and other sections later on.

```suggestion
#### CloudFormation Parameters in a Multi-Organization ADF Setup

```

The CloudFormation Parameter generation feature is fully compatible with 
the [Multi-Organization ADF Setup](./multi-organization-guide.md) approach. 

> Let's assume that we have a three AWS Org setup with a dev, int and prod 
> AWS Organization. This implies that the SSM param `/adf/org/stage` will have 
> one of the following three values: `[dev, int, prod]`; depending on the AWS 
> Organization you are in. Let's further assume that your application in scope 
> requires AWS Organization specific parameters.In that case, the `params` should 
> have the following content: 

Reviewer: ```suggestion

For example, in a setup with three AWS Organizations; with a separate
`dev`, an `int`, and a `prod` AWS Organization. This implies that the
SSM parameter `/adf/org/stage` will have one of the following three
values: `dev`, `int`, or `prod`; depending on the AWS organization 
you are in. Let's further assume that your application in scope 
requires AWS Organization specific parameters. In that case,
the `params` folder should  have the following content: 
```

```txt
params
└───global_dev.yml
└───global_int.yml
└───global_prod.yml
└───global.yml
```

> When the application gets deployed, ADF will choose the right parameter file 
> based on the value of the SSM prameter "/adf/org/stage".
> 

Reviewer: ```suggestion
Where it will prefer the AWS Organization specific configuration
`global_${org_stage}` over the `global` parameters in case they both
match the same parameter or tag.
```
